public/app/features/dashboard/components/GenAI/GenAIService.ts

```diff
@@ -0,0 +1,88 @@
+import { PanelModel } from '@grafana/data';
+import { llms } from '@grafana/experimental';
+
+import { getDashboardSrv } from '../../services/DashboardSrv';
+
+// TODO: Replace this approach with more stable approach
+export const SPECIAL_DONE_TOKEN = '~';
```

Comment:
This file definitely needs some work - we should move away from hacky special done token in favor of receiving done result from llm plugin API directly

Also we should consider a better way to share this code betwen title / description generation + potentially more use cases (at the very least use an `enum` instead of passing around "title" and "description" everywhere

Reply:
I would not concentrate all possible "Generative AI calls" in a single place. Instead, every component will contain the prompt to be sent. Then this service will become just a simplification of the llm librarly API, which it creates a signal of: Are we duplicating with no reason? Can we just modify the llm library if we find it hard to use?

IMO, we just should use llm directly with no services in between.

Reply:
I should be able to move the odd logic around detecting if streams are done into the grafana/experimental package so that we don't have to do this oddball logic with it.